.github/workflows/kafka_demo_test.yml

```diff
@@ -0,0 +1,138 @@
+        description: "Consumer Image"
+        default: acc/samples/kafka/consumer:1.0
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        default: acc/samples/kafka/producer:1.0
+        required: true
+        type: string
+      kubernetes-version:
+        description: "Kubernetes version to use for AKS cluster creation"
+        required: true
+        default: "1.28.3"
+        type: string
+  merge_group:
+    branches:
+      - main
+
+jobs:
+  prepare:
+    name: Extract Environment Variables
+    runs-on: ubuntu-latest
+    outputs:
+      sidecar-registry: ${{ steps.extract-envs.outputs.sidecar-registry }}
+      key-release-image: ${{ steps.extract-envs.outputs.key-release-image }}
+      consumer-image: ${{ steps.extract-envs.outputs.consumer-image }}
+      producer-image: ${{ steps.extract-envs.outputs.producer-image }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Extract Envs 
+        id: extract-envs
+        run: | 
+          echo "Extracting Environment Variables"
+          echo "Triggering event is ${{ github.event_name }}"
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "sidecar-registry=mcr.microsoft.com" >> $GITHUB_OUTPUT
+            echo "key-release-image=aci/skr:2.7" >> $GITHUB_OUTPUT
+            echo "consumer-image=acc/samples/kafka/consumer:1.0" >> $GITHUB_OUTPUT
+            echo "producer-image=acc/samples/kafka/producer:1.0" >> $GITHUB_OUTPUT
+          else
+            echo "sidecar-registry=${{ inputs.sidecar-registry }}" >> $GITHUB_OUTPUT
+            echo "key-release-image=${{ inputs.key-release-image }}" >> $GITHUB_OUTPUT
+            echo "consumer-image=${{ inputs.consumer-image }}" >> $GITHUB_OUTPUT
+            echo "producer-image=${{ inputs.producer-image }}" >> $GITHUB_OUTPUT
+          fi
+
+  create-prepare-aks-cluster:
+    name: Create AKS Cluster
+    needs: [prepare]
+    uses: ./.github/workflows/_create_prepare_aks_cluster.yml
+    secrets: inherit
+ 
+  deploy-kafka-test: 
+    name: Deploy Kafka Demo Test 
+    uses: ./.github/workflows/_deploy_kafka_test.yml
+    needs: [prepare, create-prepare-aks-cluster]
+    secrets: inherit 
+    with: 
+      cluster-name: ${{ needs.create-prepare-aks-cluster.outputs.cluster-name }}
+      sidecar-registry: ${{ needs.prepare.outputs.sidecar-registry }}
+      key-release-image: ${{ needs.prepare.outputs.key-release-image }}
+      consumer-image: ${{ needs.prepare.outputs.consumer-image }}
+      producer-image: ${{ needs.prepare.outputs.producer-image }}
+
+  test-workload: 
+    name: Test Workload 
+    uses: ./.github/workflows/_test_workload.yml
+    needs: [prepare, create-prepare-aks-cluster, deploy-kafka-test]
+    secrets: inherit 
+    with: 
+      cluster-name: ${{ needs.create-prepare-aks-cluster.outputs.cluster-name }}
+
+  cleanup: 
+    name: Clean Up
+    if: always()
```

Comment:
Will this error out if the cluster creation fails?

Reply:
yes, if the cluster creation fails, then the az aks delete operation would fail too.